### PR TITLE
Remove `mobile/compare` page and references

### PR DESCRIPTION
**Pull request by reemhamz**
## One-line summary
The mobile compare page is out-of-date, and we've decided to remove the page alongside any references and assets

## Significant changes and points to review
- [x] Removed `browsers/mobile/compare` page and `browsers/mobile/compare.ftl` file
- [x] Removed `browsers/mobile/compare` page reference in `browsers/mobile`
- [x] Removed banner on `firefox/browsers/compare` page
- [x] Removed assets and links leading to the page
- [x] Redirected the `browsers/mobile/compare` page to `browsers/mobile` landing page

## Issue / Bugzilla link
#14141 

## Testing
- http://localhost:8000/en-US/firefox/browsers/mobile/
- http://localhost:8000/en-US/firefox/browsers/compare

### CODE REVIEW

File: bedrock/firefox/templates/firefox/browsers/compare/index.html [hunk 1]
```diff
@@ -21,10 +21,6 @@
   {% include 'firefox/browsers/compare/includes/subnav.html' %}
 {% endblock %}
 
-{% block page_banner %}
-  {% include 'includes/banners/mobile/mobile-compare.html' %}
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/browsers/compare/index.html:25]: Question: there's a CSS bundle just above here `{{ css_bundle('mobile-compare-banner') }}` which might be able to be removed also?

Comment (alexgibson) [reply bedrock/firefox/templates/firefox/browsers/compare/index.html:25]: Oh there's a JS bundle called `mobile_compare_banner` which can be removed as well?

**Review by alexgibson — APPROVED** [review]
A couple of loose ends that can also be removed, but otherwise this looks good to me. r+wc 📞

File: bedrock/firefox/templates/firefox/browsers/compare/index.html (hunk 1 above)
Comment (reemhamz) [reply bedrock/firefox/templates/firefox/browsers/compare/index.html:25]: Thanks for the keen eye!

**Pull request merged**